### PR TITLE
use async/await for clarity

### PR DESCRIPTION
**Pull request by benma**
Less indentation, less try..catch, and should be easier to follow,
refactor and maintain the code.

### CODE REVIEW

File: src/bitbox02.js [hunk 1]
```diff
@@ -89,6 +89,20 @@ export class BitBox02API {
...
         this.devicePath = devicePath;
         this.opened = false;
+        this.socket = null;
+    }
+
+    connectWebsocket() {
+        const socket = new WebSocket("ws://127.0.0.1:8178/api/v1/socket/" + this.devicePath);
+        return new Promise((resolve, reject) => {
+            socket.binaryType = 'arraybuffer'
+            socket.onopen = async function (event) {
```
Comment (thisconnect) [inline src/bitbox02.js:99]: I think the `async` here is not needed, as it uses normal `Promise`

File: src/bitbox02.js [hunk 2]
```diff
@@ -89,6 +89,20 @@ export class BitBox02API {
...
+                resolve(socket);
+            }
+            socket.onerror = function(event) {
+                reject("Your BitBox02 is busy");
+            }
```
Comment (thisconnect) [inline src/bitbox02.js:104]: I think we don't use Automatic Semicolon Insertion (ASI), please add `;` here and after `socket.binaryType = 'arraybuffer'`

File: src/bitbox02.js [hunk 3]
```diff
@@ -99,87 +113,72 @@ export class BitBox02API {
...
+        this.socket = await this.connectWebsocket();
+        this.socket.onclose = function(event) {
+                onCloseCb();
+            }
+        this.socket.onmessage = event => {
+            this.firmware().js.OnRead(new Uint8Array(event.data));
+        }
+        const onWrite = bytes => {
+            if (this.socket.readyState != WebSocket.OPEN) {
+                console.log("Error, trying to write to closed socket");
```
Comment (thisconnect) [inline src/bitbox02.js:127]: Use single quotes

File: src/bitbox02.js [hunk 4]
```diff
@@ -99,87 +113,72 @@ export class BitBox02API {
...
+                throw new Error("Pairing rejected");
+            case constants.Status.Unpaired:
+                await userVerify()
+                await this.firmware().js.AsyncChannelHashVerify(true);
+                break;
+            case constants.Status.Initialized:
+                // Pairing skipped.
+                break;
+            default:
+                throw new Error("Unexpected status: " + this.firmware().Status() + "," + constants.Status.Unpaired);
```
Comment (thisconnect) [inline src/bitbox02.js:180]: Single quote or backtick:
`Unexpected status: ${this.firmware().Status()},${constants.Status.Unpaired}`

File: src/bitbox02.js [hunk 5]
```diff
@@ -99,87 +113,72 @@ export class BitBox02API {
...
             }
         });
+
+        await this.firmware().js.AsyncInit();
+        switch(this.firmware().Status()) {
+            case constants.Status.PairingFailed:
+                this.socket.close();
+                throw new Error("Pairing rejected");
+            case constants.Status.Unpaired:
+                await userVerify()
```
Comment (thisconnect) [inline src/bitbox02.js:173]: ASI

**Review by thisconnect — CHANGES_REQUESTED** [review]

File: src/bitbox02.js (hunk 1 above)
Comment (benma) [reply src/bitbox02.js:99]: inside the function we use `await`, so this function is `async`. didn't work removing the keyword here.

File: src/bitbox02.js (hunk 3 above)
Comment (benma) [reply src/bitbox02.js:127]: out of scope, i just moved existing code around - there are lots of other double quotes too.

would be great if we could add some linting so this is part of the process, not part of reviews.

File: src/bitbox02.js (hunk 4 above)
Comment (benma) [reply src/bitbox02.js:180]: i am just moving code around ;) these improvements are for another PR.

File: src/bitbox02.js (hunk 5 above)
Comment (benma) [reply src/bitbox02.js]: done

File: src/bitbox02.js (hunk 1 above)
Comment (benma) [reply src/bitbox02.js:99]: sorry i looked at the wrong branch - you are right, fixed

**Review by thisconnect — APPROVED** [review]
nice, untested LGTM

**Pull request merged**